runtimes/dart-2.18/server.dart

```diff
@@ -10,56 +10,121 @@ import 'function_types.dart';
-      return shelf.Response(500, body: 'Unauthorized');
+
+    String bodyString = await req.readAsString();
+    dynamic body = bodyString;
+    String method = req.method;
+    Map<String, dynamic> headers = {};
+
+    for (MapEntry entry in req.headers.entries) {
+      String header = entry.key.toLowerCase();
+      if(!header.startsWith('x-open-runtimes-')) {
+        headers[header] = entry.value;
+      }
+    }
+
+    String contentType = req.headers['content-type'] ?? 'plain/text';
+    if(contentType.contains('application/json')) {
+      if(!bodyString.isEmpty) {
+        body = jsonDecode(bodyString);
+      } else {
+        body = {};
+      }
+    }
+
+    String hostHeader = req.headers['host'] ?? '';
+
+    String path = '/' + req.url.path;
+    String host = '';
+    String scheme = req.headers['x-forwarded-proto'] ?? 'http';
+    String defaultPort = scheme == 'https' ? '443' : '80';
+    int port = int.parse(defaultPort);
+    String queryString = req.url.query;
+    Map<String, String> query = {};
+
+    if(hostHeader.contains(':')) {
+      host = hostHeader.split(':')[0];
+      port = int.parse(hostHeader.split(':')[1]);
+    } else {
+      host = hostHeader;
+      port = int.parse(defaultPort);
+    }
+
+    for(final param in queryString.split("&")) {
+      final parts = param.split("=");
+
+      final key = parts[0];
+      final value = parts.sublist(1).join('=');
+
+      if(key != null && !key.isEmpty) {
+        query[key] = value;
+      }
+    }
+
+    String url = scheme + '://' + host;
+
+    if(port != int.parse(defaultPort)) {
+      url += ':' + port.toString();
     }
-    final headers = {
-      'content-type': 'application/json'
-    };
-    
-    try {
-      final bodystring = await req.readAsString();
-      final body = jsonDecode(bodystring);
-      final request = Request(
-        variables: body['variables'] ?? {},
-        headers: body['headers'] ?? {},
-        payload: body['payload'] ?? '',
-      );
 
-      final response = Response();
-      await runZonedGuarded(
+    url += path;
+
+    if(!queryString.isEmpty) {
+      url += '?' + queryString;
+    }
+
+    Request contextReq = new Request(bodyString: bodyString, body: body, headers: headers, method: method, url: url, path: path, port: port, scheme: scheme, host: host, queryString: queryString, query: query);
+    Response contextRes = new Response();
+    Context context = new Context(contextReq, contextRes);
+
+    String customstd = "";
+
+    dynamic output = null;
+    try {
+      await runZoned(
         () async {
-          await user_code.start(request, response);
+          output = await user_code.main(context);
         },
-        (e, stackTrace) => print('$e $stackTrace'),
         zoneSpecification: ZoneSpecification(
           print: (Zone self, ZoneDelegate parent, Zone zone, String line) {
-            userLogs.add(line);
+            customstd += line;
           },
         ),
       );
```

Comment:
We are, and if we spot that you used it, we will ignore what you printed, and print "Use context.log()" instead. It helps devs quickly figure out why native way does not work.

Reply:
I really don't think this is a good idea. This will make it impossible for people to use use the standard logs if they want to.

Reply:
This might be topic where someone will always complain, no matter which way we go.
I would prefer goign with this solution as it forces developer to start using `context.log`, and provide us any feedback if this method miss-behaves.

If we get too many complains, we can remove this logic in future.